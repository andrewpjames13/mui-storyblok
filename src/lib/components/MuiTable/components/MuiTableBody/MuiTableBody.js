import React, { createElement } from 'react';
import { TableBody } from '@material-ui/core';
import PropTypes from 'prop-types';
import Storyblok from '../../../../utils/Storyblok';

import MuiTableRow from '../MuiTableRow/MuiTableRow';

const MuiTableBody = ({ rootClass, content }) => {
  const components = {
    MuiTableRow,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <TableBody className={styles.root}>
      {content.map((item, index) => createElement(
        components[item.component],
        Object.assign(item, { key: index }),
      ))}
    </TableBody>
  );
};

export default MuiTableBody;

MuiTableBody.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /**
   * Content passed to render
   * components:
    MuiTableRow,
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiTableBody.defaultProps = {
  rootClass: [],
};
