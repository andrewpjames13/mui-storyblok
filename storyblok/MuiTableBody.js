module.exports = {
  "name": "MuiTableBody",
  "display_name": null,
  "updated_at": "2020-05-06T14:54:15.192Z",
  "schema": {
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTableRow"
      ],
      "required": true
    },
    "rootClass": {
      "type": "options",
      "description": "   * stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component.\n"
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiTableBody",
  "component_group_uuid": null
}