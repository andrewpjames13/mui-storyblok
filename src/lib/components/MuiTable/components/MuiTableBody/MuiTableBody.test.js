import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiTableBody from './MuiTableBody';

function setup() {
  const props = {
    content: [{
      hover: false,
      content: [{
        size: 'medium',
        align: 'center',
        content: [{
          align: 'inherit',
          color: 'inherit',
          width: '100%',
          height: '100%',
          content: [{
            text: 'table body 1',
            component: 'MuiText',
          }],
          variant: 'body1',
          component: 'MuiTypography',
        }],
        padding: 'default',
        component: 'MuiTableCell',
      }],
      selected: false,
      component: 'MuiTableRow',
    }],
  };
  const comp = shallow(<MuiTableBody {...props} />);
  return { comp, props };
}

describe('<MuiTableBody />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiTableBody {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiBody', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
