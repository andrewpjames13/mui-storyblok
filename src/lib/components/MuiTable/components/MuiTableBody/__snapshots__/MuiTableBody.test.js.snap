// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiTableBody /> snapshot 1`] = `
<tbody
  className="MuiTableBody-root makeStyles-root-2"
  role={null}
>
  <tr
    className="MuiTableRow-root makeStyles-root-4"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body makeStyles-root-10 MuiTableCell-alignCenter"
    >
      <div
        className="MuiTypography-root makeStyles-root-23 MuiTypography-body1 MuiTypography-colorInherit"
      >
        <span
          className="MuiTypography-root makeStyles-root-24 MuiTypography-body1"
        >
           table body 1 
        </span>
      </div>
    </td>
  </tr>
</tbody>
`;
